# /home/lim/move_to_target/src/move_to_target/config/my_map_waypoints.yaml

waypoints:
  - name: "원점"
    pose:
      position: {x: 0.0, y: 0.0}
      yaw: 0
    neighbors:

      - name: "4번"
        arrival_yaw: 90   # 원점 → 4번 이동 시 도착 yaw
      - name: "7번"
        arrival_yaw: 0



  - name: "1번"
    pose:
      position: {x: 1.19, y: 0.6}
      yaw: 0
    neighbors:
      - name: "2번"
        arrival_yaw: 90
      - name: "14번"
        arrival_yaw: 180
  - name: "2번"
    pose:
      position: {x: 1.2, y: 1.509}
      yaw: 180
    neighbors:
      - name: "3번"
        arrival_yaw: 180
      - name: "1번"
        arrival_yaw: 180

  - name: "3번"
    pose:
      position: {x: 1.09, y: 2.4}
      yaw: 200
    neighbors:
      - name: "2번"
        arrival_yaw: 270
      - name: "8번"
        arrival_yaw: 90


  - name: "4번"
    pose:
      position: {x: -0.032, y: 1.43}
      yaw: 270
    neighbors:
      - name: "원점"
        arrival_yaw: 0     # ✅ 원점 → 4번일 때
      - name: "5번"
        arrival_yaw: 0    # ✅ 5번 → 4번일 때
      - name: "7번"
        arrival_yaw: 270   # ✅ 7번 → 4번일 때

  - name: "5번"
    pose:
      position: {x: -0.06, y: 2.47}
      yaw: 270
    neighbors:
      - name: "4번"
        arrival_yaw: 270
      - name: "12번"
        arrival_yaw: 0

  # - name: "6번"
  #   pose:
  #     position: {x: 1.58, y: 0.35}
  #     yaw: 200
  #   neighbors:
  #     - name: "3번"
  #       arrival_yaw: 200
  #     - name: "2번"
  #       arrival_yaw: 90


  - name: "7번"
    pose:
      position: {x: 0.1, y: 0.641}
      yaw: 270
    neighbors:
      - name: "원점"
        arrival_yaw: 90
      - name: "4번"
        arrival_yaw: 90
      - name: "14번"
        arrival_yaw: 0

  - name: "8번"
    pose:
      position: {x: 0.65, y: 2.723}
      yaw: 0
    neighbors:
      - name: "9번"
        arrival_yaw: 270
      - name: "2번"
        arrival_yaw: 270

  # - name: "9번"
  #   pose:
  #     position: {x: 0.77, y: 2.501}
  #     yaw: 0
  #   neighbors:
  #     - name: "1번"
  #       arrival_yaw: 270
  #     - name: "8번"
  #       arrival_yaw: 90
  #     - name: "10번"
  #       arrival_yaw: 270
  #     - name: "12번"
  #       arrival_yaw: 180
  #     - name: "13번"
  #       arrival_yaw: 270

  - name: "10번"
    pose:
      position: {x: 0.77, y: 2.05}
      yaw: 0
    neighbors:
      - name: "9번"
        arrival_yaw: 90

  - name: "11번"
    pose:
      position: {x: 0.43, y: 2.723}
      yaw: 0
    neighbors:
      - name: "12번"
        arrival_yaw: 270

  - name: "12번"
    pose:
      position: {x: 0.43, y: 2.501}
      yaw: 0
    neighbors:
      - name: "11번"
        arrival_yaw: 90
      - name: "13번"
        arrival_yaw: 270
      - name: "9번"
        arrival_yaw: 0
      - name: "5번"
        arrival_yaw: 270

  - name: "13번"
    pose:
      position: {x: 0.43, y: 2.233}
      yaw: 0
    neighbors:
      - name: "12번"
        arrival_yaw: 90
      - name: "9번"
        arrival_yaw: 0

  - name: "14번"
    pose:
      position: {x: 0.8, y: 0.53}
      yaw: 0
    neighbors:
      - name: "1번"
        arrival_yaw: 90
      - name: "7번"
        arrival_yaw: 180

  # - name: "입고"
  #   pose:
  #     position: {x: 0.623, y: 0.559}
  #     yaw: 180
  #   neighbors:
  #     - name: "2번"
  #       arrival_yaw: 90
  #     - name: "7번"
  #       arrival_yaw: 180
